### PR TITLE
BL-994-ds-implement-database-changes-for-meetings

### PR DESCRIPTION
**Pull request by gpwolfe**
## Description
Per discussion with @JamesFincher, endpoints have been created for the following:
* POST for create meeting  
* ~PUT~ PATCH for update meeting  
* DELETE for delete meeting
* method for returning all meetings (using empty query, as with read/mentee and read/mentor)
* method for returning meetings for specified mentor or mentee profile id, or other identifying information (i.e., meeting id, as with read/mentee and read/mentor)
  
The following changes were made to schema:
- mentor_id and mentee_id become optional in meeting update, as meeting_id is the defining key.
  
Other considerations:
- Endpoints in meeting_router.py follow existing format from mentee/mentor router endpoints.
- Tests are included in `tests/proper_integration_tests.ipynb` following existing test format.
- Version is bumped by 0.0.1

## Jira Ticket
[Jira ticket](https://bloomtechlabs.atlassian.net/jira/software/c/projects/BL/boards/10?modal=detail&selectedIssue=BL-994)
## Type of Change
- [X] New feature

## Checklist
- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

## Loom Video
[Loom video](https://www.loom.com/share/aad206c3d5944071b153b89cfe32b499)

### CODE REVIEW

**Review by nikolayglushetskiy — APPROVED** [review]
Big work!

File: app/routers/meeting_router.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @param update_data: JSON[MeetingUpdate]
+    @return JSON[Boolean] - Indicates success or failure of update
+    </pre></code>"""
+    data = update_data.dict(exclude_none=True, exclude_unset=True)
+    return {"result": Router.db.update(
+        "Meetings", {"meeting_id": meeting_id}, data)}
+
+
+@Router.delete("/delete/meeting")
+async def delete_meeting(meeting_id: str):
```
Comment (BrokenShell) [inline app/routers/meeting_router.py:55]: Let's add the id parameter to the url, like so:
`@Router.delete("/delete/meeting/{meeting_id}")`

*The feedback delete should be like this also*

File: app/api.py [hunk 2]
```diff
@@ -46,5 +47,11 @@ async def collections():
...
 
 
-for router in (mentor_router, mentee_router, feedback_router, graph_router, model_router, match_router):
+for router in (mentor_router,
+               mentee_router,
+               feedback_router,
+               graph_router,
+               model_router,
+               match_router,
+               meeting_router):
```
Comment (TophertheCoder) [inline app/api.py:56]: a lot more lines but definitely easier to read

**Review by TophertheCoder — APPROVED** [review]
this is great stuff. I especially like that both update and delete both require a specific id's to avoid deleting all or accidentally updating everything. Nice work

File: app/routers/meeting_router.py (hunk 1 above)
Comment (gpwolfe) [reply app/routers/meeting_router.py:55]: Sounds good; I can make the delete feedback change in the current archive feedback issue

File: app/routers/meeting_router.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    Displays all meetings if no input provided
+    <pre><code>
+    @param data: JSON[Optional[Dict]]
+    @return JSON[Array[Meeting]]
+    </pre></code>"""
+    return {"result": Router.db.read("Meetings", data)}
+
+
+@Router.put("/update/meeting/{meeting_id}")
+async def update_meeting(meeting_id: str, update_data: MeetingUpdate):
```
[the remark this answers is not in the document]
Comment (gpwolfe) [reply app/routers/meeting_router.py:42]: I believe patch allows partial updates/changes without having to also define the unchanged values, while put will look at MeetingUpdate (in this case) for default values where you haven't defined them in the update request, potentially overwriting values you intended to keep. This is from [the FastAPI explanation](https://fastapi.tiangolo.com/tutorial/body-updates/).

**Review by TophertheCoder — APPROVED** [review]
previous review still applies. Great work!

File: app/routers/meeting_router.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (BrokenShell) [reply app/routers/meeting_router.py:42]: PUT needs to be able to either take a full data object and add it to the database or update it if it exists already, PATCH just updates a record. The update method on the MongoDB interface is all about the PATCH way.

[the remark this answers is not in the document]
Comment (BrokenShell) [reply app/routers/meeting_router.py:42]: Just to be clear, either paradigm could work for various projects. The suggestion above matches code we already have a bit closer.

**Review by nikolayglushetskiy — APPROVED** [review]

**Pull request merged**